### PR TITLE
Milestone2 -  Create Components and Store

### PR DESCRIPTION
**Pull request by anewman15**
In this PR, we created the main components and initiated the redux store with a reducer for books. Steps include
- creating folder structure
- adding BooksList component that holds a table of books
- adding Book component that renders a book in a table row
- adding BooksForm component that creates a new book
- defining actions and reducer for creating and deleting books
- Initialize the store and provide it to App component

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello Team 👋

**You've obviously done pretty well here trying your best to follow most of the requirements. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/containers/BooksList.js
Comment (mikenath223) [inline src/containers/BooksList.js]: - [x] Kindly connect this component to the Redux store using mapStateToProps to retrieve book objects from the store.
- [x] Also, please ensure that a book object is passed as a prop to each Book component. So consider mapping through the book array retrieved from the store.

**Comment by mikenath223** [comment]
- [x] In `src/reducers/index.js`, kindly combine app specific reducers.

**Comment by visva-dev** [comment]
@mikenath223  Thank you for the review, but the changes that you requested are not from this milestone but from milestone 3 which we are working on right now.

File: src/components/Book.js
Comment (teekaytech) [inline src/components/Book.js]: The `Book` component should accept **Book object** as a prop, not id, title, and category ([screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/AccceptBookObjectAsProp.png))

File: src/containers/BooksForm.js
Comment (teekaytech) [inline src/containers/BooksForm.js]: All commented blocks of code should be removed.

File: src/containers/BooksList.js
Comment (teekaytech) [inline src/containers/BooksList.js]: - The `BookList` component should be connected to Redux store and fetches books from the state ([screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/ConnetToRedux.png))

- This component should also use mapStateToProps to fetch books from the store

- Lastly, you should pass a book object as a prop to each Book component. You do not have to break down the properties of the book. Kindly pass the book object.

These are features in Milestone 2, as seen in the project specification.
save

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone. 
However, some changes are required before it can be approved.__

_Kindly note that most of the required changes by the previous reviewer should be made in milestone 2. Please check my comments below and fix them as appropriate._

**[ REQUIRED CHANGES ]**

- [x] The `src/reducers/index.js` cannot be empty: it should combine app-specific reducers ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CombinesAppReducers.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/)). The content might be similar to the snippet below: 
```JavaScript
// src/reducers/index.js

import { combineReducers } from 'redux'
import books from './books'

export default combineReducers({
  books
})

```
- [x] Please check my inline comments below for other changes.



* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

File: src/reducers/books.js
Comment (bellom) [inline src/reducers/books.js:4]: ### [Optional]
Kindly be aware that you can use uuidv4 to automatically generate ids. read more 👉 [npm uuid](https://www.npmjs.com/package/uuidv4)

File: src/reducers/books.js
Comment (bellom) [inline src/reducers/books.js:2]: ### Optional 
These books should be actually be created in `src/index.js`
image

It would be nice to correct this, tag as optional cause it wasn't mention earlier.

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**
